core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -135,7 +135,7 @@ final class JdbcUtil {
           + CATALOG_NAME
           + " = ? AND "
           + TABLE_NAMESPACE
-          + " LIKE ? LIMIT 1";
+          + " = ?";
```

Comment:
Oh, actually we would still want the limit 1 to shortcut for the existence checks since this queries the "all tables" table.